File: backend/app/schemas/schemes.py
from pydantic import BaseModel, Field


class SchemeRecommendRequest(BaseModel):
    disease: str = Field(min_length=2, max_length=120)
    state: str = Field(min_length=2, max_length=100)
    income_category: str = Field(min_length=2, max_length=80)
    age: int = Field(ge=0, le=120)


class EligibleScheme(BaseModel):
    name: str
    benefit: str
    eligibility_reason: str
    required_documents: list[str]
    application_steps: list[str]
    nearest_help_center: str


class SchemeRecommendResponse(BaseModel):
    eligible_schemes: list[EligibleScheme]
    summary: str
    disclaimer: str

File: frontend/lib/schemes.ts
import { apiClient } from "@/lib/api-client";

export type SchemeRecommendRequest = {
  disease: string;
  state: string;
  income_category: string;
  age: number;
};

export type EligibleScheme = {
  name: string;
  benefit: string;
  eligibility_reason: string;
  required_documents: string[];
  application_steps: string[];
  nearest_help_center: string;
};

export type SchemeRecommendResponse = {
  eligible_schemes: EligibleScheme[];
  summary: string;
  disclaimer: string;
};

export async function recommendSchemes(
  payload: SchemeRecommendRequest,
): Promise<SchemeRecommendResponse> {
  return apiClient<SchemeRecommendResponse>("/api/schemes/recommend", {
    errorMessage: "Unable to recommend schemes",
    method: "POST",
    body: payload,
  });
}
